webclient/webclient/src/main/java/io/helidon/webclient/RedirectInterceptor.java

```diff
@@ -52,14 +62,17 @@ public void handleInterception(HttpResponse httpResponse,
             } else {
                 requestBuilder.uri(newUri);
             }
-            CompletionStage<WebClientResponse> redirectResponse = requestBuilder.request();
+            CompletionStage<WebClientResponse> redirectResponse =
+                    entity == null ? requestBuilder.request() : requestBuilder.submit(entity);
```

Comment:
If I am not mistaken, you are sending currently entity in all the cases. Since if we are not keeping the method, but we had entity before, you will stil propagate it and send it.